### PR TITLE
inifile : forgotten commit of an previous PR

### PR DESCRIPTION
**Pull request by guilpier-code**
See #2064

### CODE REVIEW

**Comment by flomnes** [comment]
@guilpier-code Can you use interactive rebase to include only the latest commit (fc0ef2f2f5f6350a64fbcc8488e126e829ce24fa) ? Currently you have lots of conflicts, and you're trying to merge many commits that have already been merged. Thank you

==> @guilpier-code  says : **Done**

File: src/libs/antares/inifile/include/antares/inifile/inifile.h [hunk 1]
```diff
@@ -43,22 +42,21 @@ class IniFile final
...
+        Property() = default;
         explicit Property(const AnyString& key);
         template<class U>
         Property(const AnyString& key, const U& value);
-        ~Property();
+        ~Property() = default;
 
-        template<class StreamT>
-        void saveToStream(StreamT& file, uint64_t& written) const;
+        void saveToStream(std::ostream& file, uint64_t& written) const;
```
Comment (a-zakir) [inline src/libs/antares/inifile/include/antares/inifile/inifile.h]: ofstream is more suitable for "file", unless file is not only a file

Comment (flomnes) [reply src/libs/antares/inifile/include/antares/inifile/inifile.h]: We use `std::stringstream` for unit tests (in-memory tests)

File: src/libs/antares/inifile/inifile.cpp [hunk 2]
```diff
@@ -48,6 +48,11 @@ inline void IniFile::Property::saveToStream(std::ostream& stream_out, uint64_t&
     stream_out << key << " = " << value << '\n';
 }
 
+void IniFile::Section::add(const Property property)
```
Comment (flomnes) [inline src/libs/antares/inifile/inifile.cpp]:
```suggestion
void IniFile::Section::add(const Property& property)
```

**Review by flomnes — CHANGES_REQUESTED** [review]
Apply formatting (clang-format)

**Review by flomnes — APPROVED** [review]

**Pull request merged**